### PR TITLE
global: addition of poetry

### PR DESCRIPTION
**Pull request by drjova**
Signed-off-by: Harris Tzovanakis <me@drjova.com>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -64,7 +64,10 @@ RUN npm install -g \
...
- pip install --no-cache-dir --upgrade setuptools && \
- pip install --no-cache-dir --upgrade wheel && \
- pip install --no-cache-dir -e .[all,xrootd] -r requirements.txt
+ENV PATH="/root/.poetry/bin:${PATH}" \
+    POETRY_VIRTUALENVS_CREATE=false
+
+RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python \
+    && poetry --version
+
+RUN poetry install -E all
```
Comment (pazembrz) [inline Dockerfile]: Maybe you should also add `-n` for no interaction.

File: docker-compose.deps.yml [hunk 2]
```diff
@@ -27,7 +27,7 @@ services:
     extends:
       file: services.yml
       service: base
-    command: pip install --editable .[all] --find-links /pip-cache --pre --requirement requirements.txt --upgrade
+    command: poetry install -E all
```
Comment (pazembrz) [inline docker-compose.deps.yml]: same

File: Dockerfile (hunk 1 above)
Comment (michamos) [inline Dockerfile]: no `xrootd` on purpose?

File: docker-compose.deps.yml [hunk 3]
```diff
@@ -27,7 +27,7 @@ services:
     extends:
```
Comment (michamos) [inline docker-compose.deps.yml]: so `docker-compose -f docker-compose.deps.yml run --rm pip` now runs `poetry` maybe also change the name of the service?

File: docker-compose.deps.yml [hunk 4]
```diff
@@ -27,7 +27,7 @@ services:
...
       file: services.yml
       service: base
-    command: pip install --editable .[all] --find-links /pip-cache --pre --requirement requirements.txt --upgrade
+    command: bash -c "pip install poetry && poetry export --without-hashes -f requirements.txt > requirements.txt && pip install --pre --requirement requirements.txt"
```
Comment (pazembrz) [inline docker-compose.deps.yml]: You should also install here inspirehep itself

**Comment by drjova** [comment]
@pazembrz updated

**Pull request merged**